_posts/2018-10-08-Pushing-Images-to-AWS.md

```diff
@@ -0,0 +1,159 @@
+
+Install the [AWS client](https://aws.amazon.com/cli/) tooling:
+
+    $ sudo yum install python2-pip
+    $ sudo pip2 install awscli
+
+Make sure you have an *Access Key ID* configured in
+[AWS IAM account manager](https://aws.amazon.com/iam/) and use that info to configure
+the AWS command line client:
+
+    $ aws configure
+    AWS Access Key ID [None]: ............
+    AWS Secret Access Key [None]: .............
+    Default region name [None]: us-east-1
+    Default output format [None]:
+
+Make sure you have an appropriate [S3 bucket](https://aws.amazon.com/s3/). We've called
+ours ```examplecomposer``` but yours must be globally unique, so you can't choose
+the same name:
+
+    $ BUCKET=composerredhat
+    $ aws s3 mb s3://$BUCKET
+
+![S3 bucket screenshot](/images/aws-s3-bucket-composerredhat.png)
+
+If you haven't already, create a ```vmimport``` S3 *Role* in *IAM* and grant it
+permissions to access *S3*.  This is how you do it from the command line:
+
+    $ printf '{ "Version": "2012-10-17", "Statement": [ { "Effect": "Allow", "Principal": { "Service": "vmie.amazonaws.com" }, "Action": "sts:AssumeRole", "Condition": { "StringEquals":{ "sts:Externalid": "vmimport" } } } ] }' > trust-policy.json
+    $ printf '{ "Version":"2012-10-17", "Statement":[ { "Effect":"Allow", "Action":[ "s3:GetBucketLocation", "s3:GetObject", "s3:ListBucket" ], "Resource":[ "arn:aws:s3:::%s", "arn:aws:s3:::%s/*" ] }, { "Effect":"Allow", "Action":[ "ec2:ModifySnapshotAttribute", "ec2:CopySnapshot", "ec2:RegisterImage", "ec2:Describe*" ], "Resource":"*" } ] }' $BUCKET $BUCKET > role-policy.json
+    $ aws iam create-role --role-name vmimport --assume-role-policy-document file://trust-policy.json
+    $ aws iam put-role-policy --role-name vmimport --policy-name vmimport --policy-document file://role-policy.json
+
+![IAM vmimport role](/images/aws-iam-vmimport-role.png)
+
+## Compose an image from the UI
+
+To compose an image in Composer, log into the *Cockpit Web Console* with your web browser.
+It's running on port ```9090``` on the VM that you're running Composer in. Use any admin
+or ```root``` Linux system credentials to log in.  Select the *Image Builder* tab.
+
+![Cockpit Composer](/images/cockpit-composer-main.png)
+
+We first have to have a *blueprint* defined. This blueprint describes what goes into the image.
+For the purposes of this example we'll use the ```example-http-server``` blueprint, which
+builds an image that contains a basic HTTP server.
+
+Click on the *Create Image* button and choose *Amazon Machine Image* from the dropdown
+to choose the *Image Type*:
+
+![Create Image AMI](/images/cockpit-composer-create-ami.png)
+
+If you click on the blueprint, you should see progress described on the Images tab:
+
+![Create Image Progress](/images/cockpit-composer-progress.png)
+
+Once it's done, download the image:
+
+![Download Image](/images/cockpit-composer-download.png)
+
+## Compose an image from the CLI
+
```

Comment:
We're going to have to change the name of the cli tool back to composer-cli since it conflicts with a php package manager by the same name.

Reply:
Fixed.